lib/spack/spack/build_systems/python.py

```diff
@@ -233,7 +233,27 @@ def install_args(self, spec, prefix):
         if ('py-setuptools' == spec.name or          # this is setuptools, or
             'py-setuptools' in spec._dependencies and  # it's an immediate dep
             'build' in spec._dependencies['py-setuptools'].deptypes):
-            args += ['--single-version-externally-managed', '--root=/']
+            args += ['--single-version-externally-managed']
+
+        # Get all relative paths since we set the root to `prefix`
+        # We query the python with which these will be used for the lib and inc
+        # directories. This ensures we use `lib`/`lib64` as expected by python.
+        python = spec['python'].package.command
+        commands = ';'.join([
+            'import distutils.sysconfig',
+            'print(distutils.sysconfig.get_python_lib(False, prefix=""))',
+            'print(distutils.sysconfig.get_python_lib(True, prefix=""))',
+            'print(distutils.sysconfig.get_python_inc(True, prefix=""))'])
```

Comment:
Minor, but wondering if this:
```suggestion
            'print(distutils.sysconfig.get_python_lib(plat_specific=False, prefix=""))',
            'print(distutils.sysconfig.get_python_lib(plat_specific=True, prefix=""))',
            'print(distutils.sysconfig.get_python_inc(plat_specific=True, prefix=""))'])
```
may improve readability.